litr/src/main/java/com/linkedin/android/litr/resample/AudioResampler.java

```diff
@@ -0,0 +1,28 @@
+package com.linkedin.android.litr.resample;
+
+import androidx.annotation.NonNull;
+import java.nio.ByteBuffer;
+
+/**
+ * Resamples audio data. See {@link UpsampleAudioResampler} or
+ * {@link DownsampleAudioResampler} for concrete implementations.
+ */
+public interface AudioResampler {
+
+    /**
+     * Resamples input audio from input buffer into the output buffer.
+     *
+     * @param inputBuffer the input buffer
+     * @param inputSampleRate the input sample rate
+     * @param outputBuffer the output buffer
+     * @param outputSampleRate the output sample rate
+     * @param channels the number of channels
+     */
+    void resample(@NonNull final ByteBuffer inputBuffer, int inputSampleRate, @NonNull final ByteBuffer outputBuffer, int outputSampleRate, int channels);
```

Comment:
I think, factory pattern would work better here to create correct resampler implementation for given source/target formats. We would need instantiate only one class.